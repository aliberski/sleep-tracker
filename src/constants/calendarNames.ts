export default {
  monthNames: [
    'Styczeń',
    'Luty',
    'Marzec',
    'Kwiecień',
    'Maj',
    'Czerwiec',
    'Lipiec',
    'Sierpień',
    'Wrzesień',
    'Październik',
    'Listopad',
    'Grudzień',
  ],
  monthNamesShort: [
    'Sty',
    'Lut',
    'Mar',
    'Kwi',
    'Maj',
    'Cze',
    'Lip',
    'Sie',
    'Wrz',
    'Paź',
    'Lis',
    'Gru',
  ],
  dayNames: [
    'Niedziela',
    'Poniedziałek',
    'Wtorek',
    'Środa',
    'Czwartek',
    'Piątek',
    'Sobota',
  ],
  dayNamesShort: [
    'Nie',
    'Pon',
    'Wt',
    'Śr',
    'Czw',
    'Pt',
    'Sob',
  ],
};
