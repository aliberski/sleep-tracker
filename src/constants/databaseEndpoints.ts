export default {
  SURVEY: 'survey',
  PROFILE: 'profile',
};
