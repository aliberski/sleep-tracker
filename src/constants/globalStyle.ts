import { Dimensions, StyleSheet } from 'react-native';
import { IconBackArrow } from 'components/Icons';

export const COLOR = {
  primary: '#6666EE',
  primary_variant: '#3700B3',
  secondary: '#03DAC6',
  secondary_variant: '#018786',
  background: '#FFFFFF',
  error: '#B00020',

  font: '#000000',
  font_light: '#FFFFFF',
  border: '#B4B4B4',
  disabled: '#B4B4B4',
};

export const PADDING = {
  primary: 10,
  small: 5,
  medium: 15,
  big: 20,
  large: 40,
};

export const BORDER = {
  primary: 1,
  small: 0.5,
  hair: StyleSheet.hairlineWidth,
};

export const FONT = {
  primary: 'Roboto-Regular',
  light: 'Roboto-Light',
  italic: 'Roboto-Italic',
  bold: 'Roboto-Bold',
  black: 'Roboto-Black',
};

export const FONT_SIZE = {
  primary: 15,
  small: 13,
  big: 20,
};

export const ICON_SIZE = {
  primary: 20,
  small: 15,
  big: 25,
  large: 40,
};

export const WINDOW_WIDTH = Dimensions.get('window').width;
export const DRAWER_WIDTH = WINDOW_WIDTH * 0.7;

export const tabBarOptions = {
  labelStyle: {
    fontFamily: FONT.primary,
    fontSize: FONT_SIZE.small,
    color: COLOR.font_light,
  },
  indicatorStyle: {
    backgroundColor: COLOR.primary_variant,
  },
  style: {
    backgroundColor: COLOR.primary,
  },
};

export const standardHeaderOptions = {
  headerTintColor: COLOR.primary,
  headerTitleStyle: {
    fontSize: FONT_SIZE.primary,
    fontFamily: FONT.primary,
    color: COLOR.font,
  },
  headerBackImage: IconBackArrow,
};

export const labelStyle = {
  fontFamily: FONT.bold,
  fontSize: FONT_SIZE.small,
  color: COLOR.font,
  paddingBottom: PADDING.small,
};

export const defaultTextStyle = {
  fontSize: FONT_SIZE.primary,
  fontFamily: FONT.primary,
  color: COLOR.font,
};

export const inputStyle = {
  ...defaultTextStyle,
  borderWidth: BORDER.primary,
  borderColor: COLOR.border,
  width: '100%',
  paddingHorizontal: PADDING.primary,
  paddingVertical: PADDING.medium,
};

export const formErrorStyle = {
  ...defaultTextStyle,
  paddingVertical: PADDING.big,
  color: COLOR.error,
};
