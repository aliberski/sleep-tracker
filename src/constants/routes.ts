enum routes {
  // Auth
  LOGIN = 'LOGIN',
  REGISTER = 'REGISTER',
  REMIND_PASSWORD = 'REMIND_PASSWORD',

  // App
  MAIN = 'MAIN',
  SETTINGS = 'SETTINGS',
  ABOUT = 'ABOUT',
  SURVEY = 'SURVEY',
  STATISTICS = 'STATISTICS',
}

export default routes;
