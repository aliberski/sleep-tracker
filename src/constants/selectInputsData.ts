export const physicalActivityData = [
  {
    label: 'Niska',
    value: 'low',
  },
  {
    label: 'Przeciętna',
    value: 'average',
  },
  {
    label: 'Duża',
    value: 'high',
  },
];
