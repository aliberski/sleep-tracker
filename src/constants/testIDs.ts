export const auth = {
  login: {
    id: 'LOGIN_ID',
    inputLogin: 'LOGIN_INPUT_LOGIN',
    inputPassword: 'LOGIN_INPUT_PASSWORD',
    buttonSubmit: 'LOGIN_BUTTON_SUBMIT',
    buttonRemind: 'LOGIN_BUTTON_REMIND',
  },
  register: {
    id: 'REGISTER_ID',
    inputEmail: 'REGISTER_INPUT_EMAIL',
    inputLogin: 'REGISTER_INPUT_LOGIN',
    inputPassword: 'REGISTER_INPUT_PASSWORD',
    buttonSubmit: 'REGISTER_BUTTON_SUBMIT',
  },
  remindPassword: {
    id: 'REMIND_PASSWORD_ID',
    inputEmail: 'REMIND_PASSWORD_INPUT_EMAIL',
    buttonSubmit: 'REMIND_PASSWORD_BUTTON_SUBMIT',
    buttonBack: 'REMIND_PASSWORD_BUTTON_BACK',
  },
};

export const app = {
  main: {
    id: 'MAIN_ID',
  }
};
