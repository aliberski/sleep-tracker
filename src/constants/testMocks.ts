export const navigationMock = {
  navigate: jest.fn(),
  dispatch: jest.fn(),
  dismiss: jest.fn(),
  goBack: jest.fn(),
  openDrawer: jest.fn(),
  toggleDrawer: jest.fn(),
  closeDrawer: jest.fn(),
  getParam: jest.fn(),
  setParams: jest.fn(),
  addListener: jest.fn(),
  push: jest.fn(),
  replace: jest.fn(),
  pop: jest.fn(),
  popToTop: jest.fn(),
  dangerouslyGetParent: jest.fn(),
  isFocused: jest.fn(),
  state: {},
};
