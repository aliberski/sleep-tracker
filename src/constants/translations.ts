export default {
  // common
  appName: 'Sleep tracker',
  back: 'Wróć',
  confirm: 'Zatwierdź',
  cancel: 'Anuluj',
  save: 'Zapisz',
  no: 'Nie',
  yes: 'Tak',
  timePlaceholder: 'Wybierz godzinę',

  // LoginScreen
  loginTitle: 'Logowanie',
  loginInputLogin: 'Login',
  loginInputPassword: 'Hasło',
  loginButtonRemind: 'Przypomnij hasło',
  loginButtonSubmit: 'Zaloguj się',

  // RegisterScreen
  registerTitle: 'Rejestracja',
  registerInputEmail: 'Email',
  registerInputLogin: 'Login',
  registerInputPassword: 'Hasło',
  registerButtonSubmit: 'Zarejestruj się',

  // RemindPasswordScreen
  remindTitle: 'PRZYPOMNIJ HASŁO',
  remindInputEmail: 'Email',
  remindButtonSubmit: 'Przypomnij hasło',

  // Drawer
  drawerLinkMain: 'Strona główna',
  drawerLinkProfile: 'Twój profil',
  drawerLinkSettings: 'Ustawienia konta',
  drawerLinkAbout: 'O aplikacji',
  drawerLinkLogout: 'Wyloguj się',

  // Profile
  profileTitle: 'Twój profil',

  // Settings
  settingsTitle: 'Ustawienia konta',
  settingsInputEmail: 'Email',
  settingsInputLogin: 'Login',
  settingsInputPassword: 'Hasło',
  settingsButtonSubmit: 'Zapisz',

  // Main
  mainTitle: 'Wybrany dzień: ',
  mainButtonSurvey: 'Wypełnij ankietę',
  mainButtonStatistics: 'Zobacz statystyki',

  // About
  aboutTitle: 'O aplikacji',

  // Survey
  surveyTitle: 'Wypełnij ankietę',
  surveyIntro:
    'Jak się spało?\nWypełnij ankietę, aby umożliwić zbadanie jakości Twojego snu.',
  surveyChangeDateButton: 'Zmień datę',
  surveyModalTitle: 'Wybierz datę',
  surveyLabelMood: 'Samopoczucie po przebudzeniu',
  surveyLabelSleep: 'O której poszedłeś spać?',
  surveyLabelWakeUp: 'O której wstałeś?',
  surveyLabelFood: 'Czy jadłeś przed snem?',
  surveyLabelAlcohol: 'Czy piłeś wczoraj alkohol?',

  // Statistics
  statisticsTitle: 'Statystyki snu',
};
