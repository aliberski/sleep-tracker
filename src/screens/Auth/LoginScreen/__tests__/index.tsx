import React from 'react';
import { navigationMock } from 'constants/testMocks';
import { LoginScreen } from '../index';
import renderer from 'react-test-renderer';

describe('LoginScreen', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(
        <LoginScreen
          navigation={navigationMock}
          testRequest={jest.fn()}
        />,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
