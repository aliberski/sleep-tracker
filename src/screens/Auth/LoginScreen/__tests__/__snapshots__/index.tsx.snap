// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginScreen renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "padding": 10,
    }
  }
  testID="LOGIN_ID"
>
  <View
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <RCTScrollView
      automaticallyAdjustContentInsets={false}
      contentContainerStyle={
        Object {
          "flexGrow": 1,
        }
      }
      contentInset={
        Object {
          "bottom": 0,
        }
      }
      enableAutomaticScroll={true}
      enableOnAndroid={true}
      enableResetScrollToCoords={true}
      extraHeight={75}
      extraScrollHeight={0}
      getScrollResponder={[Function]}
      handleOnScroll={[Function]}
      keyboardDismissMode="interactive"
      keyboardOpeningTime={250}
      keyboardShouldPersistTaps="handled"
      keyboardSpace={0}
      onScroll={[Function]}
      resetKeyboardSpace={[Function]}
      scrollEnabled={true}
      scrollEventThrottle={1}
      scrollForExtraHeightOnAndroid={[Function]}
      scrollIntoView={[Function]}
      scrollToEnd={[Function]}
      scrollToFocusedInput={[Function]}
      scrollToPosition={[Function]}
      showsVerticalScrollIndicator={true}
      style={
        Object {
          "backgroundColor": undefined,
          "padding": undefined,
        }
      }
      update={[Function]}
      viewIsInsideTabBar={false}
    >
      <View>
        <View>
          <Text
            style={
              Object {
                "color": "#000000",
                "fontFamily": "Roboto-Bold",
                "fontSize": 13,
                "paddingBottom": 5,
              }
            }
          >
            Login
          </Text>
          <TextInput
            allowFontScaling={true}
            onChangeText={[Function]}
            rejectResponderTermination={true}
            selectionColor="#6666EE"
            style={
              Object {
                "borderColor": "#B4B4B4",
                "borderWidth": 1,
                "fontFamily": "Roboto-Regular",
                "fontSize": 15,
                "paddingHorizontal": 10,
                "paddingVertical": 15,
                "width": "100%",
              }
            }
            testID="LOGIN_INPUT_LOGIN"
            underlineColorAndroid="transparent"
            value=""
          />
          <Text
            style={
              Object {
                "color": "#B00020",
                "fontFamily": "Roboto-Regular",
                "fontSize": 13,
                "paddingBottom": 5,
                "paddingTop": 5,
              }
            }
          />
        </View>
        <View>
          <Text
            style={
              Object {
                "color": "#000000",
                "fontFamily": "Roboto-Bold",
                "fontSize": 13,
                "paddingBottom": 5,
              }
            }
          >
            Hasło
          </Text>
          <TextInput
            allowFontScaling={true}
            onChangeText={[Function]}
            rejectResponderTermination={true}
            secureTextEntry={true}
            selectionColor="#6666EE"
            style={
              Object {
                "borderColor": "#B4B4B4",
                "borderWidth": 1,
                "fontFamily": "Roboto-Regular",
                "fontSize": 15,
                "paddingHorizontal": 10,
                "paddingVertical": 15,
                "width": "100%",
              }
            }
            testID="LOGIN_INPUT_PASSWORD"
            underlineColorAndroid="transparent"
            value=""
          />
          <Text
            style={
              Object {
                "color": "#B00020",
                "fontFamily": "Roboto-Regular",
                "fontSize": 13,
                "paddingBottom": 5,
                "paddingTop": 5,
              }
            }
          />
        </View>
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
              "padding": 5,
              "width": "100%",
            }
          }
          testID="LOGIN_BUTTON_REMIND"
        >
          <Text
            style={
              Object {
                "color": "#000000",
                "fontFamily": "Roboto-Bold",
                "fontSize": 13,
                "textAlign": "center",
              }
            }
          >
            Przypomnij hasło
          </Text>
        </View>
      </View>
    </RCTScrollView>
  </View>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "#6666EE",
        "opacity": 1,
        "paddingHorizontal": 10,
        "paddingVertical": 15,
        "width": "100%",
      }
    }
    testID="LOGIN_BUTTON_SUBMIT"
  >
    <Text
      style={
        Object {
          "color": "#FFFFFF",
          "fontFamily": "Roboto-Bold",
          "fontSize": 15,
          "textAlign": "center",
        }
      }
    >
      Zaloguj się
    </Text>
  </View>
</View>
`;
